File: 2.linked-list/single-linked-list/remove-nth-from-end/removeNthFromEnd.js
let removeNthFromEnd = function (head, n) {
  let slow = head;
  let fast = head;
  for (let i = 0; i < n; ++i) {
    if (!fast) return head;
    fast = fast.next;
  }
  while (fast && fast.next) {
    slow = slow.next;
    fast = fast.next;
  }
  slow.next = slow.next.next;
  return head;
};

File: 2.linked-list/single-linked-list/remove-nth-from-end/remove_nth_from_end.py
def removeNthFromEnd(self, head, n):
  low = fast  = head
  for _ in range(n):
    if not fast:
      return head
    fast = fast.next
  while fast and fast.next:
    slow, fast = slow.next, fast.next
  slow.next = slow.next.next
  return slow